dagify/converter/report_generator.py

```diff
@@ -23,11 +43,110 @@ def __init__(
         self.config = {}
         self.templates = {}
         self.source_path = source_path
-        self.output_path = output_path
+        source_xml_name = self.source_path.split("/")[-1].split(".")[0]
+        self.output_path = f"{output_path}/{source_xml_name}"
         self.templates_path = templates_path
 
         # Run the Proccess
-        self.generate_report()
+        self.load_source()
```

Comment:
I don't think we should copy this large chunk of code from engine.py. Ideally, we execute this code exactly once during the conversion and the results are used both for the conversion as well as for the code generation. If this is infeasible in the current code structure, then moving this chunk of code into a utilities file would make more sense.